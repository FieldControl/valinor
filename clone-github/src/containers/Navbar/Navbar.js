import React from "react";
import "./Navbar.scss";
import cloud from "../../assets/cloud.png";

/**
 * Navbar Component 
 *
 * @component
 * @example

 * <Navbar  />
 */
const Navbar = () => {
  return (
    <nav className="nav">
      <div className="nav__left">
        <img
          className="nav__icon"
          width="40px"
          height="40px"
          src={cloud}
          alt="icon"
        />
        <span className="nav__title">Github Clone</span>
      </div>
      <ul className="nav__right">
        <li>
          <a href="/" alt="home">
            clean
          </a>
        </li>
      </ul>
    </nav>
  );
};

export default Navbar;
