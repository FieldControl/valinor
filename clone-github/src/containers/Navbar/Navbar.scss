.nav__title {
  box-sizing: border-box;
  color: white;
  padding: 5px;
  font-size: 1rem;
  margin-left: 5px;
  letter-spacing: 1px;
}
.nav {
  box-sizing: border-box;
  padding: 5px;
  display: flex;
  align-items: center;
  background-color: rgb(56, 56, 56);
  width: 100%;
  font-family: Lato, sans-serif;
}
.nav__left {
  display: flex;
  align-items: center;
  width: 30%;
}
.nav__right {
  list-style: none;
  display: flex;
  justify-content: space-around;
  width: 70%;
  letter-spacing: 1px;
  & a {
    color: white;
    text-decoration: none;
    &:hover {
      opacity: 0.7;
    }
  }
}
.nav__icon {
  margin-left: 10px;
}

@media (max-width: 820px) {
  .nav__left {
    width: 50%;
  }
  .nav__right {
    width: 50%;
  }
}
