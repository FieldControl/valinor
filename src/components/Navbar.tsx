import { useState } from 'react'
import Input from './Input'
import { useRouter } from 'next/router';

export default function Navbar() {
    const [search, setSearch] = useState("");

    const router = useRouter()

    const handlesubmit = (e: React.FormEvent<HTMLFormElement>) => {
        e.preventDefault();

        if (!search) return;

        router.push(`/search?q=${search}`)
    }

    return (
            <form onSubmit={handlesubmit} className='flex flex-col sm:flex-row items-center gap-4 '>
                <Input name='input' id='input' placeholder='Pesquise no Github...' onChange={(e) => setSearch(e.target.value)} value={search} />
                <button
                    type="submit"
                    className="bg-header px-5 rounded-md font-semibold flex items-center gap-3 pointer-events-auto hover:bg-gray-800 h-[46px]"
                >
                    Buscar
                </button>

            </form>
    )
}