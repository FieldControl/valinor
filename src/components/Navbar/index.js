import { Layout, Menu } from "antd"

const { Header } = Layout

function NavBar() {
  return (
    <div>
      <Layout className="layout">
        <Header>
          <Menu theme="dark" mode="horizontal">
            Seleção
          </Menu>
        </Header>
      </Layout>
    </div>
  )
}

export default NavBar
