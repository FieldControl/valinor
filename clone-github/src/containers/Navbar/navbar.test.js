import { render, cleanup } from "@testing-library/react";
import Navbar from "./Navbar.js";

afterEach(cleanup);

test("should take a snapshoot", () => {
  const { asFragment } = render(<Navbar />);
  expect(asFragment(<Navbar />)).toMatchSnapshot();
});
