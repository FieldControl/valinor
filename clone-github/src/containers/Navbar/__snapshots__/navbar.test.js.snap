// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshoot 1`] = `
<DocumentFragment>
  <nav
    class="nav"
  >
    <div
      class="nav__left"
    >
      <img
        alt="icon"
        class="nav__icon"
        height="40px"
        src="cloud.png"
        width="40px"
      />
      <span
        class="nav__title"
      >
        Github Clone
      </span>
    </div>
    <ul
      class="nav__right"
    >
      <li>
        <a
          alt="home"
          href="/"
        >
          clean
        </a>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
